ibc-testkit/src/context.rs

```diff
@@ -0,0 +1,523 @@
+            .host
+            .latest_block()
+            .into_header()
+            .into_consensus_state()
+            .into();
+
+        let ibc_commitment_proof = self
+            .multi_store
+            .get_proof(
+                self.host.latest_height().revision_height().into(),
+                &self
+                    .ibc_store
+                    .commitment_prefix()
+                    .as_bytes()
+                    .try_into()
+                    .expect("valid utf8 prefix"),
+            )
+            .expect("no error");
+
+        self.ibc_store.begin_block(
+            self.host.latest_height().revision_height(),
+            consensus_state,
+            ibc_commitment_proof,
+        );
+    }
+
+    pub fn end_block(&mut self) {
+        // commit ibc store
+        let ibc_store_commitment = self.ibc_store.end_block().expect("no error");
+
+        // commit ibc store commitment in multi store
+        self.multi_store
+            .set(
+                self.ibc_store
+                    .commitment_prefix()
+                    .as_bytes()
+                    .try_into()
+                    .expect("valid utf8 prefix"),
+                ibc_store_commitment,
+            )
+            .expect("no error");
+    }
+
+    pub fn produce_block(&mut self, block_time: Duration, params: &H::BlockParams) {
+        // commit the multi store
+        let multi_store_commitment = self.multi_store.commit().expect("no error");
+        // generate a new block
+        self.host
+            .advance_block(multi_store_commitment, block_time, params);
+    }
+
+    pub fn advance_with_block_params(&mut self, block_time: Duration, params: &H::BlockParams) {
+        self.end_block();
+        self.produce_block(block_time, params);
+        self.begin_block();
+    }
+
+    pub fn advance_block(&mut self) {
+        self.advance_with_block_params(
+            Duration::from_secs(DEFAULT_BLOCK_TIME_SECS),
+            &Default::default(),
+        )
+    }
+
+    pub fn prune_block_till(&mut self, height: &Height) {
+        self.host.prune_block_till(height);
+        self.ibc_store.prune_host_consensus_states_till(height);
+    }
+
+    pub fn latest_height(&self) -> Height {
+        let latest_ibc_height = self.ibc_store.host_height().expect("Never fails");
+        let latest_host_height = self.host.latest_height();
+        assert_eq!(
+            latest_ibc_height, latest_host_height,
+            "The IBC store and the host chain must have the same height"
+        );
+        latest_ibc_height
+    }
+
+    pub fn latest_timestamp(&self) -> Timestamp {
+        self.host.latest_block().timestamp()
+    }
+
+    pub fn timestamp_at(&self, height: Height) -> Timestamp {
+        self.host
+            .get_block(&height)
+            .expect("block exists")
+            .timestamp()
+    }
+
+    pub fn with_client_state(mut self, client_id: &ClientId, client_state: AnyClientState) -> Self {
```

Comment:
I think the comment would be `Bootstraps a client state with this context.`

Because, this adds a client state without the IBC client creation handshake. Does that sound good to you?

Reply:
d8e65c61